src/tests/containerizer/docker_cni_tests.cpp

```diff
@@ -0,0 +1,326 @@
+                ip link set $VETH1 netns $NETNS
+                ip netns exec $NETNS ip addr add "${IP_ADDR}" dev $VETH1
+                ip netns exec $NETNS ip link set $VETH1 up
+
+                echo "{"
+                echo "  \"ip4\": {"
+                echo "    \"ip\": \"${IP_ADDR}\""
+                echo "  },"
+                echo "  \"dns\": {"
+                echo "    \"nameservers\": [ \"8.8.8.8\" ]"
+                echo "  }"
+                echo "}"
+                ;;
+            "DEL"*)
+                # $VETH0 on host network namespace will be removed automatically.
+                # If the plugin can't destroy the veth pair, it will be destroyed
+                # with the container network namespace.
+                ip link delete $VETH0 2>/dev/null || true
+                ip netns exec $NETNS ip link del $VETH1 || true
+                ;;
+            esac
+            )~").get());
+
+        ASSERT_SOME(result);
+
+        // Generate the mock CNI config for veth CNI plugin.
+        ASSERT_SOME(os::mkdir(cniConfigDir));
+
+        result = os::write(
+            path::join(cniConfigDir, MESOS_MOCK_CNI_CONFIG),
+            R"~(
+            {
+            "name": "__MESOS_TEST__",
+            "type": "mockPlugin"
+            })~");
+    }
+
+    // Generate the mock CNI plugin based on the given script.
+    Try<Nothing> setupMockPlugin(const string& pluginScript)
+    {
+        return setupPlugin(pluginScript, "mockPlugin");
+    }
+
+    // Generate the CNI plugin based on the given script.
+    Try<Nothing> setupPlugin(const string& pluginScript, const string& pluginName)
+    {
+      Try<Nothing> mkdir = os::mkdir(cniPluginDir);
+      if (mkdir.isError()) {
+        return Error("Failed to mkdir '" + cniPluginDir + "': " + mkdir.error());
+      }
+
+      string pluginPath = path::join(cniPluginDir, pluginName);
+
+      Try<Nothing> write = os::write(pluginPath, pluginScript);
+      if (write.isError()) {
+        return Error("Failed to write '" + pluginPath + "': " + write.error());
+      }
+
+      // Make sure the plugin has execution permission.
+      Try<Nothing> chmod = os::chmod(
+          pluginPath,
+          S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
+
+      if (chmod.isError()) {
+        return Error("Failed to chmod '" + pluginPath + "': " + chmod.error());
+      }
+
+      return Nothing();
+    }
+
+    static string containerName(const ContainerID& containerId)
+    {
+      return slave::DOCKER_NAME_PREFIX + containerId.value();
+    }
+
+protected:
+    slave::Flags CreateSlaveFlags() override
+    {
+        slave::Flags flags = MesosTest::CreateSlaveFlags();
+
+        flags.network_cni_plugins_dir = cniPluginDir;
+        flags.network_cni_config_dir = cniConfigDir;
+
+
+        return flags;
+    }
+};
+
+// This test tests the functionality of the docker's interfaces.
+TEST_F(DockerCniTest, ROOT_DOCKER_ip_interface)
+{
+    SetUp();
```

Comment:
nit: inconsistent indentation in your patch, can you sweep it to use a 2 space indent?